tests/components/sensor/test_nextbus.py

```diff
@@ -0,0 +1,237 @@
+"""The tests for the nexbus sensor component."""
+import unittest
+from unittest.mock import patch
+from copy import deepcopy
+
+import homeassistant.components.sensor as sensor
+import homeassistant.components.sensor.nextbus as nextbus
+
+from tests.common import (assert_setup_component,
+                          get_test_home_assistant,
+                          setup_component,
+                          MockDependency)
+
+
+VALID_AGENCY = 'sf-muni'
+VALID_ROUTE = 'F'
+VALID_STOP = '5650'
+VALID_AGENCY_TITLE = 'San Francisco Muni'
+VALID_ROUTE_TITLE = 'F-Market & Wharves'
+VALID_STOP_TITLE = 'Market St & 7th St'
+SENSOR_ID_FULL = 'sensor.san_francisco_muni_f_market_wharves'
+SENSOR_ID_SHORT = 'sensor.sf_muni_f'
+
+
+CONFIG_BASIC = {
+    'sensor': {
+        'platform': 'nextbus',
+        'agency': VALID_AGENCY,
+        'route': VALID_ROUTE,
+        'stop': VALID_STOP,
+    }
+}
+
+
+CONFIG_INVALID_MISSING = {
+    'sensor': {
+        'platform': 'nextbus',
+    }
+}
+
+
+class TestNextBusDepartureSensor(unittest.TestCase):
```

Comment:
Please use standalone pytest test functions. See the deconz test suite for example.